packages/core/src/http-middlewares/before/prepare-request.js

```diff
@@ -60,8 +60,6 @@ const coerceBody = req => {
   } else if (Buffer.isBuffer(req.body)) {
     // leave a buffer alone!
   } else if (req.body && !_.isString(req.body)) {
-    normalizeEmptyBodyFields(req);
```

Comment:
I think we still need to prune empty body fields. Can we add this back?

Reply:
@eliangcs the tricky thing here now is that we apply all templates first. Now we'll never get `{{unresolved_curly}}` by the time. 

I'm remembering the reason for the order of middleware now.

If we apply all templates (even without values) then by the time we're ready to prune values from body/params, we have already replaced everything with an empty string. For instance, we should have a different behavior for a user supplied empty string and an empty string from an unresolved curly. 

I removed `normalizeEmptyBodyFields` because we allowed empty strings to exist in the body (as we assumed they were user supplied and valid) since we removed unresolved curlies. Now, we resolve all curlies first.

In that case, would it make sense to go back to the old flow of pruning and/or resolving in the same middleware step similar to the first commit?

Reply:
> For instance, we should have a different behavior for a user supplied empty string and an empty string from an unresolved curly.

I thought that was the intended behavior. But I know the main user of `removeMissingValuesFrom` is VB. We need a way to only prune empty values that were caused by unresolved curlies because users can't delete those themselves if they're using Form Mode. Am I understanding it correctly?

> In that case, would it make sense to go back to the old flow of pruning and/or resolving in the same middleware step similar to the first commit?

If so, then yes, let's go back with the old flow. Maybe add a comment to clarify why we do it. Sorry for the hassle 🙇